client/hostvolumemanager/host_volume_plugin_test.go

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package hostvolumemanager
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-version"
+	cstructs "github.com/hashicorp/nomad/client/structs"
+	"github.com/shoenig/test"
+	"github.com/shoenig/test/must"
+)
+
+func TestHostVolumePluginMkdir(t *testing.T) {
+	volID := "test-vol-id"
+	tmp := t.TempDir()
+	target := filepath.Join(tmp, volID)
+
+	plug := &HostVolumePluginMkdir{
+		ID:         "test-mkdir-plugin",
+		TargetPath: tmp,
+		log:        hclog.Default(),
```

Comment:
There's a `helper/testlog` package that has some nice features like only emitting logs on failure (or `go test -v`) and including the location in the log outputs:

```suggestion
		log:        testlog.Logger(t),
```

Unfortunately it doesn't have a recorder, which would have been handy for your other test below.

Reply:
looks like I need a `testlog.HCLogger(t)` here, but I like!